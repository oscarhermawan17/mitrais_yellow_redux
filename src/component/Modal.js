import React from 'react'

class Modal extends React.Component{
    render(){
        return (
            <div className={this.props.modal}>
                <div className="modal-content">
                Description: <input type="text" value={this.props.form_todo.description} name="name" onChange={(e) => this.props.onChangeValueTodo(e.target.value, "description")} /> <br/>
                Deadline :
                <br/>
                <button onClick={this.props.create_single_todo}>Create Todo</button>
                <button onClick={this.props.cancel_modal}>Cancel</button>
                </div>
            </div>
        )
    }
}


export default Modal
