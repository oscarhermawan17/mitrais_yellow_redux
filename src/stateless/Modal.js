import React from 'react'

const Modal = (props) =>
(
    <div className={props.modal}>
        <div className="modal-content">
        <table>
            <tbody>
                <tr>
                    <td>Description</td>
                    <td><input type="text" value={props.form_todo.description} name="name" onChange={(e) => props.onChangeValueTodo(e.target.value, "description")} /> </td>
                </tr> 
                <tr>
                    <td>Deadline </td>
                    <td>{props.children}</td>
                </tr>
                {props.choose_page === "upd" ? 
                    <tr>                               
                        <td>Is Done ? </td>
                        <td>
                            <input type="radio" name="done" value={false} onChange={(e) => props.onChangeValueTodo(false, "done")} checked={props.form_todo.done === false} /> Not Yet<br/>
                            <input type="radio" name="done" value={true}  onChange={(e) => props.onChangeValueTodo(true, "done")} checked={props.form_todo.done === true}/> Done
                        </td>
                    </tr> : null }
            </tbody>
        </table>
            {props.choose_page === 'upd' ? 
            <button onClick={() => props.create_and_update_todo("upd")} className="btn">Update Todo</button> : 
            <button onClick={() => props.create_and_update_todo("cre")} className="btn">Create Todo</button>}
            
            <button onClick={props.cancel_modal} className="btn">Cancel</button>
        </div>
    </div>
)

export default Modal
