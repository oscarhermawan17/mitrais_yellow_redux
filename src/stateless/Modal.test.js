import { shallow } from 'enzyme'
import React from 'react'
import Modal from './Modal'

describe("Modal with Props", () => {
    const mockModal = {
        modal:"modal",
        form_todo:{
            id:"",
            description:"",
            deadline:new Date(),
            done:true,
        },
    }

    it("snapshot", ({modal, form_todo} = mockModal) => {
        expect(shallow(<Modal modal={modal} form_todo={form_todo}/>)).toMatchSnapshot()
    })
})