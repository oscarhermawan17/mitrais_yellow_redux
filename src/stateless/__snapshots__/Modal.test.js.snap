// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal with Props snapshot 1`] = `ShallowWrapper {}`;
